### PR TITLE
[quant] quantized GLU and HardGLU

### PR DESCRIPTION
**Pull request by z-a-f**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#42443 [quant] quantized GLU and HardGLU**
* #42728 HardGLU implementation
* #42727 Factoring out common code in the GLU
* #42442 [quant] Quantization parameters in TensorIterator

Differential Revision: [D23011802](https://our.internmc.facebook.com/intern/diff/D23011802)

### CODE REVIEW

File: aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp [hunk 1]
```diff
@@ -2595,60 +2692,74 @@ void dequantize_tensor_per_channel_affine_cpu(
...
-REGISTER_DISPATCH(qrelu6_stub, &qrelu6_kernel);
-REGISTER_DISPATCH(qrelu_leaky_stub, &leaky_qrelu_out_kernel);
-REGISTER_DISPATCH(qsigmoid_stub, &qsigmoid_kernel);
-REGISTER_DISPATCH(qhardsigmoid_stub, &qhardsigmoid_kernel);
-REGISTER_DISPATCH(qclamp_stub, &qclamp_kernel);
-REGISTER_DISPATCH(qthreshold_stub, &qthreshold_kernel);
-REGISTER_DISPATCH(qtanh_stub, &qtanh_kernel);
-REGISTER_DISPATCH(qhardswish_stub, &qhardswish_kernel);
-REGISTER_DISPATCH(qelu_stub, &qelu_kernel);
+REGISTER_DISPATCH(
```
Comment (z-a-f) [inline aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp:2695]: Sorry -- this is just sorting alphabetically

File: aten/src/ATen/native/quantized/cpu/quantized_ops.h [hunk 2]
```diff
@@ -152,35 +154,36 @@ using qnormalize_fn = void (*)(
...
-DECLARE_DISPATCH(qbinary_fn, qadd_stub);
-DECLARE_DISPATCH(qbinary_fn, qadd_relu_stub);
-DECLARE_DISPATCH(qbinary_fn, qmul_stub);
-DECLARE_DISPATCH(qbinary_fn, qmul_relu_stub);
-DECLARE_DISPATCH(qadd_scalar_fn, qadd_scalar_stub);
-DECLARE_DISPATCH(qadd_scalar_fn, qadd_scalar_relu_stub);
-DECLARE_DISPATCH(qhardswish_fn, qhardswish_stub);
-DECLARE_DISPATCH(qelu_fn, qelu_stub);
-DECLARE_DISPATCH(qmaxpool_2d_fn, qmaxpool_2d_nhwc_stub);
+DECLARE_DISPATCH(activation_fn, qglu_stub);
```
Comment (z-a-f) [inline aten/src/ATen/native/quantized/cpu/quantized_ops.h:157]: Sorting here again -- was really hard to find a kernel in the list

**Review by vkuzo — COMMENTED** [review]
makes sense, should we add a test as well?

File: aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp [hunk 3]
```diff
@@ -700,6 +700,103 @@ void qthreshold_kernel(
...
+
+  AT_DISPATCH_QINT_TYPES(iter.dtype(), "qhardglu", [&] {
+    using qVec = Vec256<scalar_t>;
+
+    cpu_kernel_vec(
+      iter,
+      [&](scalar_t qa, scalar_t qb) -> scalar_t {
+        const auto a = at::native::dequantize_val(a_sc, a_zp, qa);
+        const auto b = at::native::dequantize_val(b_sc, b_zp, qb);
+        const auto y = a * std::min(std::max(b + 3.0f, 0.0f), 6.0f) / 6.0f;
```
Comment (vkuzo) [inline aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp:782]: nit: we can use `three` and `six`?

**Review by vkuzo — APPROVED** [review]

**Review by vkuzo — COMMENTED** [review]
can we also add a benchmark?

**Review by raghuramank100 — APPROVED** [review]

**Pull request closed without merging**